rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -1505,9 +1505,9 @@ extern "C" rmw_ret_t rmw_publish(
   rmw_publisher_allocation_t * allocation)
 {
   static_cast<void>(allocation);    // unused
-  RET_NULL(publisher);
+  RMW_CHECK_ARGUMENT_FOR_NULL(publisher, RMW_RET_INVALID_ARGUMENT);
   RET_WRONG_IMPLID(publisher);
```

Comment:
@Lobotuerk consider changing this one to `RMW_CHECK_TYPE_IDENTIFIERS_MATCH` too.

Reply:
Isn't ```RET_WRONG_IMPLID``` doing that?

Reply:
Oh, but that macro is RMW side, not specific implementation one, makes sense, nvm